File: frontend/src/application/use-cases/users/delete-user.js
import { IUserRepository } from '../../../domain/repositories/user-repository.js';

/**
 * Caso de uso para eliminar un usuario en el frontend.
 */
export class DeleteUserUseCase {
  /**
   * @param {IUserRepository} userRepository - Una implementación del repositorio de usuarios.
   */
  constructor(userRepository) {
    if (!(userRepository instanceof IUserRepository)) {
      throw new Error('userRepository must be an instance of IUserRepository');
    }
    this.userRepository = userRepository;
  }

  /**
   * Ejecuta el caso de uso para eliminar un usuario.
   * @param {number} userId - El ID del usuario.
   * @returns {Promise<void>} Una promesa que resuelve cuando el usuario ha sido eliminado.
   */
  async execute(userId) {
    return this.userRepository.deleteOnlyUsers(userId);
  }
}

File: backend/users/application/use_cases/delete_user.py
from ...domain.repositories.user_repository import IUserRepository

class DeleteUserUseCase:
    """
    Caso de uso para eliminar un usuario.
    Esta clase reside en la capa de Aplicación.
    """
    def __init__(self, user_repository: IUserRepository):
        self.user_repository = user_repository

    async def execute(self, user_id: int) -> bool:
        # El repositorio debería tener un método delete
        # Necesito añadirlo a IUserRepository y DjangoUserRepository
        return await self.user_repository.delete(user_id)
